src/bosh_openstack_cpi/lib/cloud/openstack/helpers.rb

```diff
@@ -123,11 +123,16 @@ def wait_resource(resource, target_state, state_method = :status, allow_notfound
         # If resource reload is nil, perhaps it's because resource went away
         # (ie: a destroy operation). Don't raise an exception if this is
         # expected (allow_notfound).
-        if with_openstack { resource.reload.nil? }
-          break if allow_notfound
+        begin
+          if with_openstack { resource.reload.nil? }
+            break if allow_notfound
+            cloud_error("#{desc}: Resource not found")
+          else
+            state =  with_openstack { resource.send(state_method).downcase.to_sym }
+          end
+        rescue Bosh::Clouds::CloudError => e
```

Comment:
@ljfranklin What is the reason for having a rescue block here?`resource.reload` returns `nil` in case of no results and not an error.
[see](https://github.com/fog/fog-core/blob/f6ef6c3d9458bc235be57100acdced264e85e52a/lib/fog/core/model.rb#L48)